src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,135 @@
+import {
+    Component,
+    Input,
+    ContentChildren,
+    QueryList,
+    Renderer2,
+    ElementRef,
+    Optional,
+    SkipSelf,
+    Output,
+    AfterViewInit,
+    AfterContentInit,
+    OnDestroy
+} from '@angular/core';
+import { Router, NavigationEnd, UrlTree } from '@angular/router';
+import { filter } from 'rxjs/operators/filter';
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    }
+})
+export class NavigationItemComponent implements AfterViewInit, AfterContentInit, OnDestroy {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link: string;
+
+    get active(): boolean {
+        if (this.link) {
+            return this._router.isActive(this.link, true);
+        }
+    }
+
+    level: number = 1;
+    indentWithoutArrow: boolean = true;
+
+    private _navigationEnd: Subscription;
+    private _childrenChanges: Subscription;
+
+    @ContentChildren(NavigationItemComponent, { descendants: true })
+    private _children: QueryList<NavigationItemComponent>;
+
+    get children(): NavigationItemComponent[] {
+        return this._children.filter(item => item !== this);
+    }    
+
+    constructor(
+        private _elementRef: ElementRef,
+        private _renderer: Renderer2,
+        @Optional()
+        @SkipSelf()
+        private _parent: NavigationItemComponent,
+        private _router: Router
+    ) {
+        this.level = _parent ? _parent.level + 1 : 1;
+
+        this._navigationEnd = _router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe(event => {              
+            this.expanded = this.hasActiveLink(this.link);
+            if (this.expanded) {
+                // Expand ancestors
```

Comment:
I don't think this is needed. All navigation items subscribe to this event, and expand as required, so this would be duplicated effort.